packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -1,42 +1,45 @@
 import { v4 as uuid } from 'uuid'
-import { WalletNotFound, IsRequired, UsedName } from 'exceptions'
+import { WalletNotFound, IsRequired, UsedName, WalletFunctionNotSupported } from 'exceptions'
 import Store from 'models/store'
 import Keystore from 'models/keys/keystore'
 import WalletDeletedSubject from 'models/subjects/wallet-deleted-subject'
 import { WalletListSubject, CurrentWalletSubject } from 'models/subjects/wallets'
 import { AccountExtendedPublicKey, DefaultAddressNumber } from 'models/keys/key'
+import { Address as AddressInterface } from "models/address"
 
 import FileService from './file'
 import AddressService from './addresses'
 import ProcessUtils from 'utils/process'
 import { ChildProcess } from 'utils/worker'
+import { AddressType } from 'models/keys/address'
 
 const fileService = FileService.getInstance()
 
 const MODULE_NAME = 'wallets'
 
-export interface Wallet {
-  id: string
+export interface DeviceProperties {
   name: string
-
-  loadKeystore: () => Keystore
-  accountExtendedPublicKey: () => AccountExtendedPublicKey
+  publicKey: string
+  addressIndex: number
+  addressType: AddressType
```

Comment:
IMHO, `addressIndex` and `addressType` can also be added as properties of `WalletProperties`, since in software wallets we need to calculate the path as well：
 https://github.com/nervosnetwork/neuron/blob/9857dc9d5e0d3a9bf605d1825a77a8d8f59edb24/packages/neuron-wallet/src/services/transaction-sender.ts#L86-L91

It might be better for `DeviceProperties` to be named `DeviceInfo`, there are some fields that need to be added to implement some functionality, here is the` DeviceInfo` data structure for my branch:
```typescript
export enum Manufacturer {
  Ledger = 'Ledger'
}

export interface DeviceInfo {
  descriptor: string
  publicKey: string
  vendorId: string
  manufacturer: Manufacturer
  product: string
  connectType: 'bluetooth' | 'usb'
  // The following information may not be available to us in some wallets
  isConnected?: boolean
  appVersion?: string
  firewareVersion?: string
}
```
But we can just ignore that for now and deal with it when I merge the code.

Reply:
> IMHO, addressIndex and addressType can also be added as properties of WalletProperties, since in software wallets we need to calculate the path as well

These two properties are for single address hardware wallet for now. For software wallet, the addresses are derived from the extended public key `extendedKey` field and the two properties are recorded in the hd_public_key_info table for each address generation.

Because the initial version of hardware wallet support is single address based, assuming they are not derived from an extended public key, I store the `addressIndex` and `addressType` in the `DeviceInfo` as a part of wallet JSON store.